docs/design/openstack/networking-infrastructure.md

```diff
@@ -0,0 +1,71 @@
+installer, however our VIPs along with our internal dns and loadbalancing is managed
+by the [machine-config-operator(MCO)](https://github.com/openshift/machine-config-operator/tree/master/docs).
+The MCO has been configured to run static pods on the Bootstrap, Master, and Worker Nodes that
+run our internal networking infrastructure. Files run on the bootstrap node can be found 
+[here](https://github.com/openshift/machine-config-operator/tree/master/manifests/openstack).
+Files run on both master and worker nodes can be found 
+[here](https://github.com/openshift/machine-config-operator/tree/master/templates/common/openstack/files).
+Files run on only master nodes can be found 
+[here](https://github.com/openshift/machine-config-operator/tree/master/templates/master/00-master/openstack/files).
+Lastly, files run only on worker nodes can be found 
+[here](https://github.com/openshift/machine-config-operator/tree/master/templates/worker/00-worker/openstack/files).
+
+## Infrastructure Walkthrough
+
+The bootstrap node is responsible for running temporary networking infrastructure while the Master
+nodes are still coming up. The bootstrap node will run a coredns instance, as well as 
+keepalived. While the bootstrap node is up, it will have priority running the API and DNS
+VIPs.
+
+The Master nodes run dhcp, haproxy, coredns, mdns publisher, and keepalived. Haproxy loadbalances incoming requests 
+to the api across all running masters. It also runs a stats and healthcheck server. Keepalived manages all 3 VIPs on the master, where each
+master has an equal chance of being assigned one of the VIPs. Be aware, the bootstrap node has the highest priority for hosting the DNS
+and API VIPs, so they will point to addresses there for as long as they can. Once the master nodes have stood up a fully functioning control plane, 
+the bootstrap node will be torn down by the installer. Keepalived implements periodic health checks to make sure
+that essential services can be reached by their VIPs, when it determines that they cannot, it alters the VIP to point to a server it deems fit.  
+To ensure the api is reachable through the API VIP, keepalived periodically attempts to reach the api through the API VIP. It will do the same 
+for the DNS VIP. This means that when the bootstrap node gets torn down, keepalived will detect it, and move the API and DNS VIPs to point to a working master node. The Ingress VIP is also managed by a set of health checks that qurey the haproxy healthcheck servers to make sure that the Ingress VIP is pointing to
+a server that is able to service it.
+
+The Worker Nodes run dhcp, coredns, mdns publisher, and keepalived. On workers, keepalived is only responsible for managing
+the Ingress VIP. Just like on masters, it will periodically check the haproxy healthcheck server to make sure the node the Ingress
```

Comment:
I think it is worth it to mention that it is the ocp router haproxy and not the infra haproxy. Also, explaining that since the ocp routers do not run on every worker node, this is a good way to have the same config only make those nodes, where an ocp router gets scheduled, eligible.

Reply:
Do you mind explaining this further?

Reply:
Sure. We do not know a priori which worker nodes will get an ocp router pod scheduled. So we run keepalived for ingress on all nodes. Only those that actually run the ocp router pod get max score, so those are the ones that will get the ingress VIP.

Reply:
also, just to clarify, by ocp router pod, are you talking about the haproxy that we run in static pods, or something like multus?